web3/providers/rpc.py

```diff
@@ -57,6 +74,94 @@ def make_request(self, method, params):
         return response_body
 
 
+class KeepAliveRPCProvider(BaseProvider):
```

Comment:
Is there a reason this doesn't inherit from the `RPCProvider`?  Looks like there is a decent amount of duplication between the two classes.